web/src/test/java/org/openmrs/module/filter/ModuleFilterDefinitionTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.filter;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.hamcrest.Matchers.*;
+
+import org.w3c.dom.Document;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import org.xml.sax.InputSource;
+import java.io.StringReader;
+
+import java.util.List;
+
+import org.openmrs.module.Module;
+import org.openmrs.module.ModuleException;
+import org.openmrs.module.web.filter.ModuleFilterDefinition;
+
+import org.junit.Test;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+@PrepareForTest(ModuleFilterDefinition.class)
+public class ModuleFilterDefinitionTest {
+	public static Document getDocument(String xmlString) {
+		try {
+			DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+			DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+			Document doc = docBuilder.parse(new InputSource(new StringReader(xmlString)));
+			return doc;
+		} catch (Exception e) {
+			return null;
+		}
+	}
+
+	/**
+	 * @see ModulefilterDefinition#retrieveFilterDefinitions
+	 * @throws ModuleException
+	 */
+	@Test(expected = ModuleException.class)
```

Comment:
It's better to use junit rules so that you can also check the error messsage

Reply:
@wluyima, my bad on that. So, from what I see over at https://github.com/junit-team/junit4/wiki/rules#expectedexception-rules, should I write something similar to that code for this test?

Reply:
There is some default ones that come with jnunit, in this case specifically you need to use `ExpectedException`

Reply:
I'll let you know when the commit is amended